test/python/conftest.py

```diff
@@ -14,101 +14,94 @@ def module_tmpdir(request):
+
+        '.rocks/bin/luatest',
+        '.rocks/share/tarantool/checks.lua',
+        '.rocks/share/tarantool/luarocks/test/luatest.lua',
+        '.rocks/share/tarantool/luatest',
+        '.rocks/share/tarantool/rocks/checks',
+        '.rocks/share/tarantool/rocks/luatest',
+    ])
 
 
 @pytest.fixture(scope="module")
-def prepare_ignore(project_path):
-    """function creates files and directories
-    to check the work .cartridge.ignore"""
-
-    def create_file(path, text=None):
-        with open(path, 'w') as f:
-            if text:
-                f.write(text)
-
-    for item in ignored_data:
-        directory = os.path.join(project_path, item['dir'])
-        if not os.path.exists(directory):
-            os.makedirs(directory)
-        create_file(os.path.join(directory, item['file']))
-
-    create_file(
-        os.path.join(project_path, ".cartridge.ignore"),
-        cartridge_ignore_text)
+def project(module_tmpdir):
+    project_name = 'original-project'
+    project_path = create_project(module_tmpdir, project_name, 'cartridge')
+
+    # add third-party module dependency to the rockspec
+    current_rockspec = None
+    with open(os.path.join(project_path, '{}-scm-1.rockspec'.format(project_name)), 'r') as f:
+        current_rockspec = f.read()
+
+    with open(os.path.join(project_path, '{}-scm-1.rockspec'.format(project_name)), 'w') as f:
+        f.write(re.sub(
+            r"dependencies = {",
+            "dependencies = {\n    'custom-module == scm-1',",
+            current_rockspec
+        ))
+
+    # create custom-module itself
+    dependency_module_path = os.path.join(project_path, 'third_party', 'custom-module')
+    os.makedirs(dependency_module_path)
+    with open(os.path.join(dependency_module_path, 'custom-module-scm-1.rockspec'), 'w') as f:
+        rockspec_lines = [
+            "package = 'custom-module'",
+            "version = 'scm-1'",
+            "source  = { url = '/dev/null' }",
+            "build = { type = 'none'}",
+        ]
+        f.write('\n'.join(rockspec_lines))
+
+    # add cartridge.pre-build file to install custom-module dependency
+    with open(os.path.join(project_path, 'cartridge.pre-build'), 'w') as f:
+        prebuild_script_lines = [
+            "#!/bin/sh",
+            "tarantoolctl rocks make --chdir ./third_party/custom-module",
+        ]
+        f.write('\n'.join(prebuild_script_lines))
+
+    # add cartridge.post-build file to remove test/ and tmp/ contents
+    with open(os.path.join(project_path, 'cartridge.post-build'), 'w') as f:
+        postbuild_script_lines = [
+            "#!/bin/sh",
+            "rm -rf test tmp third_party"
+        ]
+        f.write('\n'.join(postbuild_script_lines))
+
+    # add custom-module to rocks content
+    rocks_content = get_rocks_content(project_name)
+    rocks_content.add('.rocks/share/tarantool/rocks/custom-module')
+
+    return {
```

Comment:
You return it as a module

Let's add a "Project" as class

Reply:
with options, methods and so on

Reply:
I like your idea, but it's not about new flow, so let's implement it in separated PR.

Reply:
don't forget that